source/Calamari.Testing/EnvironmentVariables.cs

```diff
@@ -70,31 +77,46 @@ public enum ExternalVariable
         [EnvironmentVariable("CALAMARI_AUTHPASSWORD", "OctopusMyGetTester")]
         MyGetFeedPassword,
 
-        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester")]
+        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester", "op://Calamari Secrets for Tests/Google Cloud Octopus Api Tester JsonKey")]
         GoogleCloudJsonKeyfile,
-        
+
         [EnvironmentVariable("GitHub_RateLimitingPersonalAccessToken", "GitHub test account PAT")]
         GitHubRateLimitingPersonalAccessToken,
     }
 
     public static class ExternalVariables
     {
+        static readonly bool SecretManagerIsEnabled = Convert.ToBoolean(Environment.GetEnvironmentVariable("CALAMARI__Tests__SecretManagerEnabled") ?? "True");
+        static readonly string SecretManagerAccount = Environment.GetEnvironmentVariable("CALAMARI__Tests__SecretManagerAccount") ?? "octopusdeploy.1password.com";
+
+        static readonly Lazy<SecretManagerClient> SecretManagerClient = new(LoadSecretManagerClient);
+
+        static SecretManagerClient LoadSecretManagerClient()
+        {
+            var allVariableAttributes = EnvironmentVariableAttribute.GetAll()
+                                                                    .Select(x => x.SecretReference)
+                                                                    .Where(x => x is not null)
+                                                                    .ToArray();
+            var loggerFactory = new LoggerFactory(); //.AddSerilog(Logger);
+            var microsoftLogger = loggerFactory.CreateLogger<SecretManagerClient>();
```

Comment:
Not really familar with how the logger should be set up, but this looks unusual at the very least.
Can I assume you've tested this and it works as expected?